python/lsst/resources/http.py

```diff
@@ -221,6 +221,37 @@ def finalurl(r: requests.Response) -> str:
     return destination_url
 
 
+# Tuple (path, block_size) pointing to the location of a local directory
+# to save temporary files and the block size of the underlying file system
+_TMPDIR = None
+
+
+def _get_temp_dir() -> Tuple[str, int]:
+    """Return the temporary directory path and block size.
+
+    This function caches its results in _TMPDIR.
+    """
+    global _TMPDIR
+    if _TMPDIR:
+        return _TMPDIR
+
+    # Use the value of environment variables 'LSST_BUTLER_TMPDIR' or
+    # 'TMPDIR', if defined. Otherwise use current working directory
+    tmpdir = os.getcwd()
+    for dir in (os.getenv(v) for v in ("LSST_BUTLER_TMPDIR", "TMPDIR")):
+        if dir and os.path.isdir(dir):
+            tmpdir = dir
```

Comment:
The implication from the code coverage is that neither of these environment variables are set. What you can do is in the test do something like:

```python
with unittest.mock.patch.dict(os.environ, {"TMPDIR": TESTDIR}):
    with self.existingFileResourcePath.as_local() as local_uri:
        ...
        self.assertIsNotNone(local_uri.relative_to(ResourcePath(TESTDIR)))
```
which would check that the local URI is a subdirectory of TESTDIR.

Reply:
It occurs to me that this is not going to be reliable because we cache the `_TEMPDIR` and so by definition are not allowing the answer to change between calls. I guess the test could force the `_TEMPDIR` global to be `None` before re-running the test (it's allowed to know about the implementation).